src/Auth0/Login/Auth0JWTUser.php

```diff
@@ -2,99 +2,35 @@
 
 namespace Auth0\Login;
 
+use Auth0\Login\Traits\Auth0UserTrait;
+use Illuminate\Contracts\Auth\Authenticatable;
+
 /**
  * This class represents a generic user initialized with the user information
  * given by Auth0 and provides a way to access to the decoded JWT data.
  */
-class Auth0JWTUser implements \Illuminate\Contracts\Auth\Authenticatable
+class Auth0JWTUser implements Authenticatable
 {
-    private $userInfo;
+
+    use Auth0UserTrait;
 
     /**
      * Auth0JWTUser constructor.
      *
-     * @param $userInfo
+     * @param array $userInfo
      */
-    public function __construct($userInfo)
+    public function __construct(array $userInfo)
     {
-        $this->userInfo = get_object_vars($userInfo);
+        $this->userInfo = $userInfo;
     }
 
     /**
      * Get the unique identifier for the user.
      *
-     * @return mixed
+     * @return string|null
      */
     public function getAuthIdentifierName()
     {
-        return $this->userInfo['sub'];
-    }
-
-    /**
-     * Get the unique identifier for the user.
-     *
-     * @return mixed
-     */
-    public function getAuthIdentifier()
-    {
-        return $this->userInfo['sub'];
-    }
-
-    /**
-     * @return void
-     */
-    public function getAuthPassword()
-    {
-    }
-
-    /**
-     * @return void
-     */
-    public function getRememberToken()
-    {
-    }
-
-    /**
-     * @param string $value
-     */
-    public function setRememberToken($value)
-    {
-    }
-
-    /**
-     * @return void
-     */
-    public function getRememberTokenName()
-    {
-    }
-
-    /**
-     * Add a generic getter to get all the properties of the userInfo.
-     *
-     * @return the related value or null if it is not set
-     */
-    public function __get($name)
-    {
-        if (!array_key_exists($name, $this->userInfo)) {
-            return;
-        }
-
-        return $this->userInfo[$name];
-    }
-
-    /**
-     * @return array
-     */
-    public function getUserInfo()
-    {
-        return $this->userInfo;
-    }
-
-    /**
-     * @return string
-     */
-    public function __toString()
-    {
-        return json_encode($this->userInfo);
+        return $this->userInfo['sub'] ?? null;
```

Comment:
I'm not so sure about this but it's, effectively, what was there before. Shouldn't this be `"sub"`? Currently (and now) it's returning the same as `getAuthIdentifier()` in the trait.